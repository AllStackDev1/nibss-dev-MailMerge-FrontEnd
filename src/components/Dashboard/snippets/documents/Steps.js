import React from 'react'
import { Step, StepDelimiter } from 'styles/styled-components/DocumentSteps';

const Steps = ({ step }) => {
    const documentStepClasses = `smooth 
        height-35 
        width-35 
        display-flex 
        align-items-center 
        justify-center 
        border-radius-100-percent`;

    const activeStepClass = 'active-step';

    return (
        <>
            <Step className={`${step === 1 && activeStepClass} ${documentStepClasses}`}>
                1
            </Step>
            <StepDelimiter></StepDelimiter>
            <Step className={`${step === 2 && activeStepClass} ${documentStepClasses}`}>
                2
            </Step>
            <StepDelimiter></StepDelimiter>
            <Step className={`${step === 3 && activeStepClass} ${documentStepClasses}`}>
                3
            </Step>
            <StepDelimiter></StepDelimiter>
            <Step className={`${step === 4 && activeStepClass} ${documentStepClasses}`}>
                4
            </Step>
        </>)
}

export default Steps;
