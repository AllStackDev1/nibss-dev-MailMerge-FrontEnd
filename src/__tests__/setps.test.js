import React from 'react';
import { shallow } from 'enzyme';
import Steps from "../components/Dashboard/snippets/documents/Steps"



const shallowSetup = (props = {}) => {

    return shallow(<Steps />)
}


it("should render without errors", () => {

    const wrapper = shallowSetup();
    expect(wrapper.find("[data-test='setup-component']").length).toBe(1)

})

it('should match snapshot', () => {
    const wrapper = shallowSetup()
    expect(wrapper).toMatchSnapshot();
})

it('should match snapshot step 2', () => {
    const wrapper = shallowSetup({step: 2})
    expect(wrapper).toMatchSnapshot();
})


it('should match snapshot step 3', () => {
    const wrapper = shallowSetup({step: 3})
    expect(wrapper).toMatchSnapshot();
})


it('should match snapshot step 4', () => {
    const wrapper = shallowSetup({step: 4})
    expect(wrapper).toMatchSnapshot();
})