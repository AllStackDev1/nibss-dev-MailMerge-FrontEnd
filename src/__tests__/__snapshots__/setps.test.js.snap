// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `ShallowWrapper {}`;

exports[`should match snapshot step 2 1`] = `ShallowWrapper {}`;

exports[`should match snapshot step 3 1`] = `ShallowWrapper {}`;

exports[`should match snapshot step 4 1`] = `ShallowWrapper {}`;
